src/snowflake/cli/api/utils/cursor.py

```diff
@@ -18,3 +18,9 @@ def find_first_row(
 ) -> Optional[dict]:
     """Returns the first row that matches the predicate, or None."""
     return next(_rows_generator(cursor, predicate), None)
+
+
+def join_cursors(cursors: List[Iterable[SnowflakeCursor]]) -> Iterable[SnowflakeCursor]:
+    while cursors:
+        cur = cursors.pop(0)
```

Comment:
nit: list needs to continually be modified, `for cur in cursors:` feels simpler and more performant given `cursors` is a `List`

Reply:
Good point, adjusted